### ISSUE TITLE
RNHealth: An error occured while trying to add sample from: <bundle id>

### ISSUE THREAD

**Issue by devth** [agencyenterprise/react-native-health#177]
**Describe the bug**
When I try to fetch workouts, I get an error message in the logs saying:

```
RNHealth: An error occured while trying to add sample from: <bundle id>
```

where `<bundle id>` is my bundle identifier.

This apparently comes from [this line](https://github.com/agencyenterprise/react-native-health/blob/c19eb40caff1b044c0c81694235421d62ebe761f/RCTAppleHealthKit/RCTAppleHealthKit%2BQueries.m#L186).

**To Reproduce**

1. Initialize permissions with `AppleHealthKit.initHealthKit` and enable them (in this case I requested all and enabled all)
2. Run code like:

```typescript
  
    const options = {
      startDate: new Date(2021, 10, 1).toISOString(),
      type: "Workout" as HealthObserver,
    };

    AppleHealthKit.getSamples(
      options,
      (err: string, results: Array<HealthValue>) => {
          console.log({err, results});
      }
    );
  });

```

**Expected behavior**

Expected to get the workouts, or an error message explaining why it couldn't instead of a generic error message.

**Smartphone (please complete the following information):**

- Device: iPhone 13 Simulator
- OS: iOS 15

```
    "react-native-health": "^1.11.0",
```

**Additional context**

I created the workout samples in the simulator by running:

```typescript
export const createWorkout = () => {
  const now = new Date();
  const start = subDays(now, 1);
  const end = addHours(start, 2);

  const options = {
    type: AppleHealthKit.Constants.Activities.Cycling,
    startDate: start.toISOString(),
    endDate: end.toISOString(),
    energyBurned: 50, // In Energy burned unit
    distance: 50, // In Distance unit
  };

  console.log("creating workout", { options });
  AppleHealthKit.saveWorkout(options, (err: string, results: HealthValue) => {
    if (err) {
      console.log("error saving workout to Healthkit: ", err);
      return;
    }
    // workout successfully saved
    console.log("workout saved", { results });
  });
};
```

I created 10 workouts this way. They show up in the Health app in the simulator:

![image](https://user-images.githubusercontent.com/86107/141343731-75c71963-34a4-431f-89a9-f55c5e39bea8.png)

**Comment by devth** [comment]
The same code runs fine on my iPhone with real workout data, so I'm guessing something about the fake workouts I generated is causing RN to throw the exception.

**Comment by gigby** [comment]
@devth Hi! I get the same errors. And I can't see my events that were added via "saveWorkout" method. But I can see these events in Healthkit app. How did you solve this issue??

**Comment by gigby** [comment]
Fixed on my master - https://github.com/Warecorp/react-native-health

**Comment by gigby** [comment]
@devth I know objective-c poorly. But my issue is not related to a physical device or simulator. This is just a logical issue in this library and I fixed it. About PR - is this library alive now? It seems abandoned

**Comment by GGGava** [comment]
Fixed by #279

**Comment by GGGava** [comment]
Fix released in [v.1.13.6](https://github.com/agencyenterprise/react-native-health/releases/tag/v1.13.6)
